x-pack/plugins/observability_solution/inventory/public/components/grouped_inventory/group_selector.tsx

```diff
@@ -0,0 +1,122 @@
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ENTITY_TYPE } from '@kbn/observability-shared-plugin/common';
+import { useInventoryPageViewContext } from '../../context/inventory_page_view_provider';
+
+const GROUP_LABELS: Record<string, string> = {
+  none: i18n.translate('xpack.inventory.groupedInventoryPage.noneLabel', {
+    defaultMessage: 'None',
+  }),
+  [ENTITY_TYPE]: i18n.translate('xpack.inventory.groupedInventoryPage.typeLabel', {
+    defaultMessage: 'Type',
+  }),
+};
+
+export interface GroupedSelectorProps {
+  groupSelected: string;
+  onGroupChange: (groupSelection: string) => void;
+}
+
+export function GroupSelector() {
+  const { euiTheme } = useEuiTheme();
+  const { grouping, setGrouping } = useInventoryPageViewContext();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const onGroupChange = (selected: string) => {
+    setGrouping(grouping === selected ? 'none' : selected);
+  };
+  const isGroupSelected = (groupKey: string) => {
+    return grouping === groupKey;
+  };
+  const panels = [
+    {
+      id: 'firstPanel',
+      title: 'Select grouping',
+      items: [
+        {
+          'data-test-subj': 'panel-none',
+          name: GROUP_LABELS.none,
+          icon: isGroupSelected('none') ? 'check' : 'empty',
+          onClick: () => onGroupChange('none'),
+        },
+        {
+          'data-test-subj': 'panel-type',
+          name: GROUP_LABELS[ENTITY_TYPE],
+          icon: isGroupSelected(ENTITY_TYPE) ? 'check' : 'empty',
+          onClick: () => onGroupChange(ENTITY_TYPE),
+        },
+      ],
+    },
+  ];
+
+  const onButtonClick = useCallback(() => setIsPopoverOpen((currentVal) => !currentVal), []);
+
+  const closePopover = useCallback(() => setIsPopoverOpen(false), []);
+
+  const button = (
+    <EuiButtonEmpty
+      data-test-subj="group-selector-dropdown"
+      css={css`
+        font-weight: 'normal';
+
+        .euiButtonEmpty__text {
+          max-width: 300px;
+          white-space: nowrap;
+          overflow: hidden;
+          text-overflow: ellipsis;
+        }
+      `}
+      flush="both"
+      iconSide="right"
+      iconSize="s"
+      iconType="arrowDown"
+      onClick={onButtonClick}
+      title={GROUP_LABELS[grouping]}
+      size="s"
+    >
+      <FormattedMessage
+        id="xpack.inventory.groupedInventoryPage.groupedByLabel"
+        defaultMessage={`Group entities by: {grouping}`}
+        values={{ grouping: GROUP_LABELS[grouping] }}
+      />
+    </EuiButtonEmpty>
+  );
+
+  return (
+    <EuiPopover
+      data-test-subj="inventory.groupedInventoryPage.groupsPopover"
+      button={button}
+      closePopover={closePopover}
+      isOpen={isPopoverOpen}
+      panelPaddingSize="none"
+    >
+      <EuiContextMenu
+        css={css`
+          width: 250px;
+          & .euiContextMenuItem__text {
+            overflow: hidden;
+            text-overflow: ellipsis;
```

Comment:
There are examples out there showing how to properly truncate text without overriding the style.